Bottom: 338bbd4e7a43c4875f6f373cb2eb623c3e6c6fdc
Top:    343beca35626921995c7272aa8b209ebfe6a8701
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-03-11 19:48:22 +0000

Remove unused "last" parameter from HID "invalidate_lr" method


---

diff --git a/src/draw.c b/src/draw.c
index b843da6..9571b29 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -194,7 +194,7 @@ Draw (void)
 
   /* clear and create event if not drawing to a pixmap
    */
-  gui->invalidate_lr (Block.X1, Block.X2, Block.Y1, Block.Y2, 1);
+  gui->invalidate_lr (Block.X1, Block.X2, Block.Y1, Block.Y2);
 
   RestoreCrosshair (True);
 
diff --git a/src/hid.h b/src/hid.h
index 0e0c479..7409bbf 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -304,8 +304,7 @@ typedef enum
     void (*parse_arguments) (int *argc_, char ***argv_);
 
     /* This may be called to ask the GUI to force a redraw of a given area */
-    void (*invalidate_lr) (int left_, int right_, int top_, int bottom_,
-			   int last_);
+    void (*invalidate_lr) (int left_, int right_, int top_, int bottom_);
     void (*invalidate_all) (void);
 
     /* During redraw or print/export cycles, this is called once per
diff --git a/src/hid/batch/batch.c b/src/hid/batch/batch.c
index 1baa2d8..f26321d 100644
--- a/src/hid/batch/batch.c
+++ b/src/hid/batch/batch.c
@@ -191,7 +191,7 @@ batch_parse_arguments (int *argc, char ***argv)
 }
 
 static void
-batch_invalidate_lr (int l, int r, int t, int b, int last)
+batch_invalidate_lr (int l, int r, int t, int b)
 {
 }
 
diff --git a/src/hid/common/hidnogui.c b/src/hid/common/hidnogui.c
index 3e5e9db..d5dc9e1 100644
--- a/src/hid/common/hidnogui.c
+++ b/src/hid/common/hidnogui.c
@@ -50,7 +50,7 @@ nogui_parse_arguments (int *argc, char ***argv)
 }
 
 static void
-nogui_invalidate_lr (int l, int r, int t, int b, int last)
+nogui_invalidate_lr (int l, int r, int t, int b)
 {
   CRASH;
 }
diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index 330e716..b8fdc75 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -318,7 +318,7 @@ ghid_get_export_options (int *n_ret)
 }
 
 void
-ghid_invalidate_lr (int left, int right, int top, int bottom, int last)
+ghid_invalidate_lr (int left, int right, int top, int bottom)
 {
   ghid_invalidate_all ();
 }
diff --git a/src/hid/lesstif/main.c b/src/hid/lesstif/main.c
index 2fec418..51c52fe 100644
--- a/src/hid/lesstif/main.c
+++ b/src/hid/lesstif/main.c
@@ -2828,9 +2828,9 @@ lesstif_need_idle_proc ()
 }
 
 static void
-lesstif_invalidate_lr (int l, int r, int t, int b, int last)
+lesstif_invalidate_lr (int l, int r, int t, int b)
 {
-  if (!last || !window)
+  if (!window)
     return;
 
   need_redraw = 1;
@@ -2840,7 +2840,7 @@ lesstif_invalidate_lr (int l, int r, int t, int b, int last)
 void
 lesstif_invalidate_all (void)
 {
-  lesstif_invalidate_lr (0, PCB->MaxWidth, 0, PCB->MaxHeight, 1);
+  lesstif_invalidate_lr (0, PCB->MaxWidth, 0, PCB->MaxHeight);
 }
 
 static int
